include/kernel/thread.h

```diff
@@ -123,7 +123,7 @@ static inline bool_t isCurDomainExpired(void)
 
 static inline void commitTime(void)
 {
-    if (NODE_STATE(ksCurSC)->scRefillMax) {
+    if (NODE_STATE(ksCurSC)->scRefillMax && (NODE_STATE(ksCurSC) != NODE_STATE(ksIdleSC))) {
```

Comment:
nit: recommend adding an `unlikely` on the idle case, since we'd prefer the compiler make that one the slowest branch.